### ISSUE TITLE
[question] Performance Problems

### ISSUE THREAD

**Issue by radonish** [conan-io/conan#12652]
My team has run into performance problems when running Conan. I'm trying to support our corporate IT team in figuring out what may be causing the problem and hoping the community can help. For example a simple Conan version check takes almost 23 seconds now:

```
$ time conan --version
Conan version 1.55.0

**real	0m22.943s**
user	0m1.088s
sys	0m0.277s
```

Doing a version check used to be almost instantaneous. 

Here is what `strace -f -c conan --version` looks like:
```
Conan version 1.55.0
% time     seconds  usecs/call     calls    errors syscall
------ ----------- ----------- --------- --------- ----------------
 28.09    0.069903          69      1003       127 open
 20.20    0.050289          16      3009       497 stat
 17.45    0.043429         700        62        27 wait4
  9.92    0.024697          17      1427           fstat
  8.47    0.021072          14      1499           read
  4.49    0.011174          17       631         7 lseek
  4.35    0.010815           9      1150        12 close
  1.46    0.003645          46        78           openat
  1.30    0.003245           5       571           mmap
  1.19    0.002957          18       156           getdents
  0.71    0.001775           6       261           mprotect
  0.59    0.001462           6       216           brk
  0.35    0.000865           1       463           rt_sigaction
  0.27    0.000662           1       613           rt_sigprocmask
  0.24    0.000587           4       121           munmap
  0.18    0.000445          74         6           getrandom
  0.17    0.000413           7        52        37 ioctl
  0.12    0.000300           1       171        83 access
  0.04    0.000094           2        35           clone
  0.04    0.000091           2        33           write
  0.03    0.000068           8         8           set_robust_list
  0.03    0.000067           2        27           rt_sigreturn
  0.03    0.000067           8         8           futex
  0.02    0.000060           8         7         1 lstat
  0.02    0.000060          20         3         2 readlink
  0.02    0.000052           2        24           pipe
  0.02    0.000050           2        24           execve
  0.02    0.000050           0        86           getuid
  0.02    0.000050           0        86           getgid
  0.02    0.000048           0        86           geteuid
  0.02    0.000047           0        86           getegid
  0.02    0.000046           1        24           arch_prctl
  0.02    0.000040           3        11           getrlimit
  0.02    0.000040          10         4           getdents64
  0.01    0.000032           4         7         1 fcntl
  0.01    0.000025           1        14        13 statfs
  0.01    0.000024          12         2           unlink
  0.01    0.000023           0        38           dup2
  0.01    0.000020          20         1           getcwd
  0.01    0.000019           3         5           uname
  0.01    0.000017           5         3           getpgrp
  0.01    0.000015           2         6           poll
  0.01    0.000015           2         6           set_tid_address
  0.00    0.000012           4         3           getpid
  0.00    0.000011           3         3           getppid
  0.00    0.000007           2         3           sigaltstack
  0.00    0.000005           1         5           pipe2
  0.00    0.000002           2         1           socket
  0.00    0.000002           2         1           bind
  0.00    0.000000           0         3           dup
  0.00    0.000000           0         1           link
  0.00    0.000000           0         1         1 fadvise64
------ ----------- ----------- --------- --------- ----------------
100.00    0.248894                 12144       808 total
```

OS: RHEL 7.9
Python Version: 3.7.1
Conan Version: 1.55.0

My apologies for the generic nature of the description but right now it seems like a generic performance issue with the execution of Conan and at least of the pip utility (pip --version has similar extremely large timings). The hope in posting the question early is that someone else has run into this within a large corporate IT environment and can help with some initial ideas. 

Any suggestions as to what to look at would be appreciated. Thank you!

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by jcar87** [comment]
Hi @radonish  thanks for raising this question.

I have done a couple tests on Linux machines but can't replicate this issue, `conan --version` is fairly instant.

My I ask how Conan itself is installed? Is this installed inside a venv with pip?

When launching the `conan` command, it will typically invoke a python interpreter (which in its own will require some libraries to be launched), and then it will load Conan's python scripts that perform the command. If you run:

```
strace -e trace=open,openat conan --version
``` 

You should see which files files are opened, typically you would expect some shared libraries to be loaded when the interpreter is launched, followed by `.py` files and/or `.pyc` files. 

to troubleshoot this further, I would suggest:
* run that command above to list which files are being opened, and in particular - check whether the files are in a filesystem that is mounted from a different block device, a network share, or something of the sort. This could be filesystem related issues
* is this reproducible on your end on multiple systems, or only one? 
* is there a setting that is inhibiting the python interpreter from creating or using the `__pycache__` folders? although in my experience this should not cause the `conan --version` command to take that long.
* is there any chance at all that the python interpreter that is running is actually built without optimisations? 
You can run `head -n 1 $(which conan)` to show which python interpreter is in use, and then invoke `readlink -f` with the returned result, in case it is a symlink to a different location on disk. Then trace wether that interpreter belongs to a system package.

**Comment by radonish** [comment]
Hello, @jcar87 - thank you for replying and my apologies for not answering your questions sooner. 

We've been debugging this and some of your questions were spot on to where the issue may be. In short, yes, multiple systems are having this slow performance problem. The Python 3 + Conan installation (not via venv) is on a NFS share used by developer machines. The current thought is that it may be related to the network route to the NFS share - that this route was changed and unknowingly has created the performance issue. By creating a new Python 3 + Conan installation on both the NFS share and my local disk I was able to prove that the extremely slow performance issue only exists on the NFS share.

Thank you
